File: assignments/solutions/01-flow-of-program/sumOfAllNumbers.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const sumOfAllNumbers = (input) => {
    return input.reduce((total, num) => {
        return Number(total) + Number(num);
    });
};

let inputArray = [],
    takeInput = true;

console.log('Sum of All Numbers\nEnter x to stop entering numbers.');

// The below function is an IIFE (Immediately Invoked Function Expression)
(function loop() {
    rl.question('Enter number: ', input => {
        // Stop input if user enters x or invalid number
        if (input === 'x' || isNaN(Number(input))) {
            takeInput = false;
            console.log(`Sum of all numbers: ${sumOfAllNumbers(inputArray)}`);
            return rl.close();
        };
        inputArray.push(input);
        takeInput = true;
        takeInput ? loop() : rl.close();
    });

})();

/**
 * Output:
 * 
 * Sum of All Numbers
 * Enter x to stop entering numbers.
 * Enter number: 3
 * Enter number: 4
 * Enter number: 5
 * Enter number: 6
 * Enter number: x
 * Sum of all numbers: 18
 */

File: assignments/solutions/03-conditionals-loops/sumOfAllNumbers.java
// Sum of All Numbers
// Take integer inputs till the user enters 0 and print the sum of all numbers

import java.util.Scanner;

public class sumOfAllNumbers {
    public static void main (String args[]) {
        int result = 0, input;
        Scanner scan = new Scanner(System.in);
        System.out.println("Sum of All Numbers\n");
        System.out.println("Keep on Entering Numbers to print their Sum");
        System.out.println("Enter -1 to stop entering numbers");
        
        while ((input = scan.nextInt()) != -1) {
            if(input == -1) break;
            result += input;
        }
        System.out.println("Sum of all Numbers = " + result);
    }
}

/*
Output:

Sum of All Numbers

Keep on Entering Numbers to print their Sum
Enter -1 to stop entering numbers
9
3
44
5
-1
Sum of all Numbers = 61
*/